### PR TITLE
wb-postback: add a timeout to allow the validation plugin to run and check for any form errors before AJAX submission

### PR DESCRIPTION
**Pull request by polmih**
Add a timeout to allow the validation plugin (wb-frmvld) to run. If there are no validation errors, the script will continue and  AJAX the submission.

As per WET-199

### CODE REVIEW

**Review by duboisp — CHANGES_REQUESTED** [review]

File: src/plugins/wb-postback/wb-postback.js [hunk 1]
```diff
@@ -43,40 +43,48 @@ var $document = wb.doc,
...
-						} )
-						.always( function() {
-
-							// Make the form submittable again if multiple submits are allowed or hide
-							if ( multiple ) {
-								$elm.removeAttr( attrEngaged );
-							} else {
-								$elm.addClass( classToggle );
+
+					// setTimeout to allow wb-frmvld plugin to generate the error section if any
```
[the remark this answers is not in the document]
Comment (polmih) [reply src/plugins/wb-postback/wb-postback.js]: "setTimeout aproach" is removed. Now the script is checking if the form is using the wet-boew form validation plugin and will allow the form to submit only if there are no form validation errors. Thanks.

**Review by GormFrank — CHANGES_REQUESTED** [review]
@polmih Really like the approach to prevent submit when jQuery validate is in use. Can you add a second form example on the page showcasing the postback plugin using the form validator (English and French)? You could use the example I made to test the bug before this PR: 
- Code: https://github.com/GormFrank/website-gcweb/blob/main/test.html
- Example: https://gormfrank.github.io/website-gcweb/test.html

\CC @duboisp

**Comment by polmih** [comment]
>Can you add a second form example on the page showcasing the postback plugin using the form validator (English and French)? You could use the example I made to test the bug before this PR:
> 
> * Code: https://github.com/GormFrank/website-gcweb/blob/main/test.html
> * Example: https://gormfrank.github.io/website-gcweb/test.html
> 
> \CC @duboisp
Done. Thank you

File: src/plugins/wb-postback/wb-postback-fr.hbs [hunk 2]
```diff
@@ -65,3 +65,62 @@
...
+<p>Pour voir cet example en action, vérifier l'onglet réseaux de la console du fureteur afin de voir la requête Ajax envoyé.</p>
+<div class="wb-frmvld">
+<form action="wb-postback-fr.html" class="wb-postback" data-wb-postback='{"success":".success-message-2","failure":".failure-message-2"}'>
+	<div class="form-content">
+		<div class="form-group">
+			<label for="firstname" class="required"><span class="field-name">Prénom</span><strong class="required">(required)</strong></label>
+			<input class="form-control" id="firstname" name="firstname" type="text" data-rule-minlength="2" required="required"/>
+		</div>
+		<div class="form-group">
+			<label for="lastname" class="required"><span class="field-name">Nom de famille</span><strong class="required">(required)</strong></label>
```
Comment (GormFrank) [inline src/plugins/wb-postback/wb-postback-fr.hbs:80]:
```suggestion
			<label for="lastname" class="required"><span class="field-name">Nom de famille</span> <strong class="required">(obligatoire)</strong></label>
```

File: src/plugins/wb-postback/wb-postback-fr.hbs [hunk 3]
```diff
@@ -65,3 +65,62 @@
...
 </details>
+
+<p>Soumission de formulaire via une requête Ajax en utilisant aussi le plugiciel <a href="https://wet-boew.github.io/wet-boew/demos/formvalid/formvalid-fr.html">Validation de formulaires</a> .</p>
+
+<p>Pour voir cet example en action, vérifier l'onglet réseaux de la console du fureteur afin de voir la requête Ajax envoyé.</p>
+<div class="wb-frmvld">
+<form action="wb-postback-fr.html" class="wb-postback" data-wb-postback='{"success":".success-message-2","failure":".failure-message-2"}'>
+	<div class="form-content">
+		<div class="form-group">
+			<label for="firstname" class="required"><span class="field-name">Prénom</span><strong class="required">(required)</strong></label>
```
Comment (GormFrank) [inline src/plugins/wb-postback/wb-postback-fr.hbs:76]:
```suggestion
			<label for="firstname" class="required"><span class="field-name">Prénom</span> <strong class="required">(obligatoire)</strong></label>
```

File: src/plugins/wb-postback/wb-postback-fr.hbs [hunk 4]
```diff
@@ -65,3 +65,62 @@
...
+<p class="failure-message-2 hide">Nous somme désolé, un problème est survenu lors de la soumission de votre formulaire. Veuillez nous envoyer l'information via une méthode alternative.</p>
+</div>
+<details class="mrgn-tp-md mrgn-bttm-md">
+	<summary>View code</summary>
+	<pre><code>
+		&lt;div class="wb-frmvld"&gt;
+		&lt;form action="wb-postback-fr.html" class="wb-postback" data-wb-postback="{&amp;quot;success&amp;quot;:&amp;quot;success-message-2&amp;quot;,&amp;quot;failure&amp;quot;:&amp;quot;failure-message-2&amp;quot;}"&gt;
+	&lt;div class="form-content"&gt;
+		&lt;div class="form-group"&gt;
+			&lt;label for="firstname" class="required"&gt;&lt;span class="field-name"&gt;Prénom&lt;/span&gt;&lt;strong class="required"&gt;(required)&lt;/strong&gt;&lt;/label&gt;
```
Comment (GormFrank) [inline src/plugins/wb-postback/wb-postback-fr.hbs:104]:
```suggestion
			&lt;label for="firstname" class="required"&gt;&lt;span class="field-name"&gt;Prénom&lt;/span&gt; &lt;strong class="required"&gt;(obligatoire)&lt;/strong&gt;&lt;/label&gt;
```

File: src/plugins/wb-postback/wb-postback-fr.hbs [hunk 5]
```diff
@@ -65,3 +65,62 @@
...
+			&lt;input class="form-control" id="firstname" name="firstname" type="text" data-rule-minlength="2" required="required"/&gt;
+		&lt;/div&gt;
+		&lt;div class="form-group"&gt;
+			&lt;label for="lastname" class="required"&gt;&lt;span class="field-name"&gt;Nom de famille&lt;/span&gt;&lt;strong class="required"&gt;(required)&lt;/strong&gt;&lt;/label&gt;
```
Comment (GormFrank) [inline src/plugins/wb-postback/wb-postback-fr.hbs:108]:
```suggestion
			&lt;label for="lastname" class="required"&gt;&lt;span class="field-name"&gt;Nom de famille&lt;/span&gt; &lt;strong class="required"&gt;(obligatoire)&lt;/strong&gt;&lt;/label&gt;
```

**Review by GormFrank — CHANGES_REQUESTED** [review]
Small text changes in French otherwise good to go!

File: src/plugins/wb-postback/wb-postback-fr.hbs (hunk 2 above)
Comment (polmih) [reply src/plugins/wb-postback/wb-postback-fr.hbs]: Fixed. Thank you!

File: src/plugins/wb-postback/wb-postback-fr.hbs (hunk 3 above)
Comment (polmih) [reply src/plugins/wb-postback/wb-postback-fr.hbs]: Fixed. Thank you!

File: src/plugins/wb-postback/wb-postback-fr.hbs (hunk 4 above)
Comment (polmih) [reply src/plugins/wb-postback/wb-postback-fr.hbs]: Fixed. Thank you!

File: src/plugins/wb-postback/wb-postback-fr.hbs (hunk 5 above)
Comment (polmih) [reply src/plugins/wb-postback/wb-postback-fr.hbs]: Fixed. Thank you!

File: src/plugins/wb-postback/wb-postback.js [hunk 6]
```diff
@@ -42,6 +42,16 @@ var $document = wb.doc,
 				// Prevent regular form submit
 				e.preventDefault();
 
+				//Check if the form use the validation plugin
+				if ( elm.parentElement && elm.parentElement.classList.contains( "wb-frmvld" ) ) {
```
Comment (duboisp) [inline src/plugins/wb-postback/wb-postback.js:46]: No need to check for `elm.parentElement`, this will be only false for the HTML element

**Review by duboisp — CHANGES_REQUESTED** [review]

File: src/plugins/wb-postback/wb-postback.js (hunk 6 above)
Comment (polmih) [reply src/plugins/wb-postback/wb-postback.js]: Fixed. Thanks

**Review by GormFrank — APPROVED** [review]

**Review by duboisp — APPROVED** [review]

**Pull request merged**